specs/wp-plan-purchase-spec.js

```diff
@@ -66,4 +67,68 @@ describe( `[${ host }] Plans: (${ screenSize }) @parallel @jetpack`, function()
 			} );
 		}
 	} );
+
+	describe( 'Viewing a specific plan with coupon:', function() {
+		let originalCartAmount, loginFlow;
+
+		before( async function() {
+			return await driverManager.ensureNotLoggedIn( driver );
+		} );
+
+		step( 'Login and Select My Site', async function() {
+			loginFlow = new LoginFlow( driver );
+			return await loginFlow.loginAndSelectMySite();
+		} );
+
+		step( 'Can Select Plans', async function() {
+			const statsPage = await StatsPage.Expect( driver );
+			await statsPage.waitForPage();
+			const sideBarComponent = await SidebarComponent.Expect( driver );
+			return await sideBarComponent.selectPlan();
+		} );
+
+		step( 'Can Select Plans tab', async function() {
+			let route = `plans/${ loginFlow.account.loginURL }`;
+			return await driver.get( dataHelper.getCalypsoURL( route ) );
+		} );
+
+		step( 'Select Business Plan', async function() {
+			const plansPage = await PlansPage.Expect( driver );
+			return await plansPage.selectBusinessPlan();
+		} );
+
+		step( 'Can Correctly Apply Coupon', async function() {
+			const planCheckoutPage = await PlanCheckoutPage.Expect( driver );
+
+			await planCheckoutPage.toggleCartSummary();
+			originalCartAmount = await planCheckoutPage.cartTotalAmount();
+
+			await planCheckoutPage.enterCouponCode( dataHelper.getTestCouponCode() );
+
+			let newCartAmount = await planCheckoutPage.cartTotalAmount();
+			let expectedCartAmount = parseFloat( ( originalCartAmount * 0.99 ).toFixed( 2 ) );
+
+			assert.strictEqual( newCartAmount, expectedCartAmount, 'Coupon not applied properly' );
+		} );
+
+		step( 'Can Remove Coupon', async function() {
+			const planCheckoutPage = await PlanCheckoutPage.Expect( driver );
+
+			await planCheckoutPage.removeCoupon();
+
+			// Needs a little time for the coupon changes to take effect
+			// Had a driver.sleep( 1000 ) here but it might it seems the time it takes to attempt to toggle twice covers it
+			await planCheckoutPage.toggleCartSummary();
+			await planCheckoutPage.toggleCartSummary();
```

Comment:
I think that because of this notice, you need to `toggleCartSummary()` twice. If you could manage to dismiss notice maybe you would need to do it only once? 
<img width="401" alt="screen shot 2018-10-28 at 23 07 12" src="https://user-images.githubusercontent.com/7116222/47622536-52308b80-db06-11e8-928c-dc8d500e9196.png">

Reply:
There's no notice that appears after removing a coupon, so that's why I decided to toggle here, to give a little time for the change to happen. But I removed them and it seems to work fine now. 
Thanks for pointing this notice out though I included dismissing it into the step after applying the coupon.